### PR TITLE
[GJ-46] clean and unify Jni interfaces

### PR DESCRIPTION
**Pull request by rui-mo**
This pr:

- added a config to decide whether to load Gandiva and Arrow libraries. Please set "spark.oap.sql.columnar.loadarrow" to false to disable loading Arrow and Gandiva libraries.

-   cleaned and unified below parts:

    ExpressionEvaluator
    ExpressionEvaluatorJniWrapper
    BatchIterator.java
    JniUtils and JniInstance
    createNativeKernelWithIterator

- removed ColumnarBroadcastExchangeExec.scala. The support for BHJ will be added in following development.
- renamed GazellePluginConfig to GazelleJniConfig.

Closes https://github.com/oap-project/gazelle-jni/issues/46.

### CODE REVIEW

File: jvm/src/main/scala/com/intel/oap/GazellePluginConfig.scala [hunk 1]
```diff
@@ -25,22 +25,22 @@ case class GazelleNumaBindingInfo(
     totalCoreRange: Array[String] = null,
     numCoresPerExecutor: Int = -1) {}
 
-class GazellePluginConfig(conf: SQLConf) extends Logging {
+class GazelleJniConfig(conf: SQLConf) extends Logging {
```
Comment (zzcclp) [inline jvm/src/main/scala/com/intel/oap/GazellePluginConfig.scala:28]: Change the file name to GazelleJniConfig.scala too?

File: jvm/src/main/scala/com/intel/oap/execution/WholestageNativeRowRDD.scala [hunk 2]
```diff
@@ -19,7 +19,7 @@ package com.intel.oap.execution
```
Comment (zzcclp) [inline jvm/src/main/scala/com/intel/oap/execution/WholestageNativeRowRDD.scala:19]: Please change the class name to 'NativeWholestageRowRDD', and don't modify the file name, make the named form the same as 'NativeWholestageColumnarRDD'.

**Comment by zzcclp** [comment]
I have tested with ClickHouse as lib module, it worked.

File: jvm/src/main/scala/com/intel/oap/execution/WholestageNativeRowRDD.scala (hunk 2 above)
Comment (rui-mo) [reply jvm/src/main/scala/com/intel/oap/execution/WholestageNativeRowRDD.scala]: Fixed.

File: jvm/src/main/scala/com/intel/oap/GazellePluginConfig.scala (hunk 1 above)
Comment (rui-mo) [reply jvm/src/main/scala/com/intel/oap/GazellePluginConfig.scala:28]: Fixed.

**Review by zzcclp — APPROVED** [review]
LGTM

**Pull request merged**